enhanced_validation block configuration ignored during terraform plan

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

Terraform v1.15.0 on darwin_arm64

### AzureRM Provider Version

4.71.0

### Affected Resource(s)/Data Source(s)

azurerm_xxx

### Terraform Configuration Files

```hcl
# Working configuration


$env:ARM_PROVIDER_ENHANCED_VALIDATION = $false

provider "azurerm" {
  features {}

  subscription_id                 = "00000000-0000-0000-0000-000000000000"
  resource_provider_registrations = "none"
}


# Failing configuration (using enhanced_validation block)


provider "azurerm" {
  features {}

  subscription_id                 = "00000000-0000-0000-0000-000000000000"
  resource_provider_registrations = "none"

  enhanced_validation {
    locations          = false
    resource_providers = false
  }
}
```

### Debug Output/Panic Output

```shell
2026-05-03T13:32:13.355+0200 [DEBUG] provider.terraform-provider-azurerm_v4.71.0_x5: GET /subscriptions/00000000-0000-0000-0000-000000000000/providers?api-version=2022-09-01 HTTP/1.1

2026-05-03T13:32:13.612+0200 [DEBUG] provider.terraform-provider-azurerm_v4.71.0_x5: {"error":{"code":"SubscriptionNotFound","message":"The subscription '00000000-0000-0000-0000-000000000000' could not be found."}}

2026-05-03T13:32:13.613+0200 [ERROR] provider.terraform-provider-azurerm_v4.71.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=88deaa62-7e0c-123e-39f5-17ccc92e9fb9 tf_rpc=Configure @module=sdk.proto diagnostic_detail="" diagnostic_summary="populating Resource Provider cache: listing Resource Providers: loading results: unexpected status 404 (404 Not Found) with error: SubscriptionNotFound: The subscription '00000000-0000-0000-0000-000000000000' could not be found." tf_proto_version=5.10 @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 timestamp="2026-05-03T13:32:13.612+0200"

2026-05-03T13:32:13.613+0200 [ERROR] vertex "provider[\"registry.terraform.io/hashicorp/azurerm\"]" error: populating Resource Provider cache: listing Resource Providers: loading results: unexpected status 404 (404 Not Found) with error: SubscriptionNotFound: The subscription '00000000-0000-0000-0000-000000000000' could not be found.
```

### Expected Behaviour
```hcl
enhanced_validation {
  resource_providers = false
}

# SHOULD prevent calls to "/subscriptions/{id}/providers"
```

### Actual Behaviour

```hcl
enhanced_validation {
  resource_providers = false
}

# DOES NOT prevent provider listing call
```

### Steps to Reproduce

terraform plan

### Important Factoids

_No response_

### References

_No response_

--- a/internal/provider/framework/config.go
+++ b/internal/provider/framework/config.go
@@ -605,11 +605,13 @@ func (p *ProviderConfig) Load(ctx context.Context, data *ProviderModel, tfVersio
 	ctx2, cancel := context.WithTimeout(ctx, 30*time.Minute)
 	defer cancel()
 
-	// Ensure that we do not trigger the RP cache when running in VCR mode or the cassettes have a base size of 3.5MiB!
-	if os.Getenv("TC_TEST_VIA_VCR") == "" {
-		if err = resourceproviders.EnsureRegistered(ctx2, client.Resource.ResourceProvidersClient, subId, requiredResourceProviders); err != nil {
-			diags.AddError("registering resource providers", err.Error())
-			return
+	if f.EnhancedValidation.ResourceProviders {
+		// Ensure that we do not trigger the RP cache when running in VCR mode or the cassettes have a base size of 3.5MiB!
+		if os.Getenv("TC_TEST_VIA_VCR") == "" {
+			if err = resourceproviders.EnsureRegistered(ctx2, client.Resource.ResourceProvidersClient, subId, requiredResourceProviders); err != nil {
+				diags.AddError("registering resource providers", err.Error())
+				return
+			}
 		}
 	}
 

--- a/internal/provider/provider.go
+++ b/internal/provider/provider.go
@@ -619,10 +619,12 @@ func buildClient(ctx context.Context, p *schema.Provider, d *schema.ResourceData
 	ctx2, cancel := context.WithTimeout(ctx, 30*time.Minute)
 	defer cancel()
 
-	// Skip this if we're running VCR, it creates too much noise in the cassette
-	if os.Getenv("TC_TEST_VIA_VCR") == "" {
-		if err = resourceproviders.EnsureRegistered(ctx2, client.Resource.ResourceProvidersClient, subscriptionId, requiredResourceProviders); err != nil {
-			return nil, diag.FromErr(err)
+	if features.EnhancedValidation.ResourceProviders {
+		// Skip this if we're running VCR, it creates too much noise in the cassette
+		if os.Getenv("TC_TEST_VIA_VCR") == "" {
+			if err = resourceproviders.EnsureRegistered(ctx2, client.Resource.ResourceProvidersClient, subscriptionId, requiredResourceProviders); err != nil {
+				return nil, diag.FromErr(err)
+			}
 		}
 	}
 

